src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1309,7 +1309,12 @@ private async Task ValidateUnusedIdsAsync(HashSet<string> unusedIds, string data
 
         public static unsafe bool IsBase64String(string base64)
         {
-            int base64Size = (int)Math.Ceiling((double)base64.Length / 3) * 4;
+            // base64string length has to be a product of 4
```

Comment:
the thing is that `Base64Id` of the database is explicitly use only 22 chars
```
public string Base64Id { get; } = new string(' ', 22);
```
So let's make this method less generic
- check for 22 chars
- always add padding and validate

something like
```
 char* buffer = stackalloc char[24];
 fixed (char* str = dbId)
 {
     Buffer.MemoryCopy(str, buffer, 24 * sizeof(char), 22 * sizeof(char));
     buffer[22] = '=';
     buffer[23] = '=';
    return Convert.TryFromBase64String(buffer, ... );
 }
```

Reply:
Done